libraries/cyclestreets-view/src/main/java/net/cyclestreets/views/overlay/Bubble.kt

```diff
@@ -0,0 +1,122 @@
+package net.cyclestreets.views.overlay
+
+import android.content.Context
+import android.content.Intent
+import android.graphics.*
+import android.net.Uri
+import android.view.MotionEvent
+import android.webkit.URLUtil
+import androidx.core.content.ContextCompat
+import net.cyclestreets.util.Draw
+import org.osmdroid.api.IProjection
+import org.osmdroid.views.MapView
+import net.cyclestreets.Undoable
+import net.cyclestreets.view.R
+import net.cyclestreets.views.CycleMapView
+import org.osmdroid.views.Projection
+
+object Bubble {
+    var activeItem: POIOverlay.POIOverlayItem? = null
+    private var bubble: Rect? = null
+    private val curScreenCoords = Point()
+    private lateinit var tapHereText: String
+    private lateinit var controller: ControllerOverlay
+
+    fun initialise(context: Context, mapView: CycleMapView) {
+        controller = mapView.controllerOverlay_
+        tapHereText = context.getString(R.string.tap_here)
+    }
+
+    fun onSingleTap(event: MotionEvent, pj: Projection, context: Context, overlay: Undoable): Boolean {
+        if (activeItem != null && tappedInBubble(event, pj, context, overlay))
+            return true
+        return false
+    }
+
+    private fun tappedInBubble(event: MotionEvent, pj: Projection, context: Context, overlay: Undoable): Boolean {
+        val screenRect = pj.intrinsicScreenRect
+        val eventX = screenRect.left + event.x.toInt()
+        val eventY = screenRect.top + event.y.toInt()
+
+        if (!bubble!!.contains(eventX, eventY))
+            return false
+        // Check if tapped on link
+        if (eventY < Draw.titleSectionY) {
+            showWebpage(activeItem, context)
+        }
+
+        hideBubble(overlay)
+        return true
```

Comment:
hmmm.  This is different behaviour from the old code in `POIOverlay.kt`, which had:

```kt
       // Check if tapped on link
        if (eventY < Draw.titleSectionY) {
            showWebpage(activeItem)
            return true
        }

        return routeMarkerAtItem(activeItem)
```
Why are we now always hiding  the bubble?

Reply:
And are we missing `routeMarkerAtItem()` sometimes?

Reply:
Only hiding the bubble if they've tapped on it.  (There is a return a few lines earlier if they tap outside the bubble.)
routeMarkerAtItem also hides the bubble.
See note 7 above as I had a question about routeMarkerAtItem. 
 If we use it, routeMarkerAtItem will only apply to POIOverlay, as CircularRoutePOIOverlay is already showing the route and waymarks can't be added.

Reply:
Note 7 makes sense.